File: src/main/webapp/src/model/bookDto.ts
export interface Book {
  id?: number;
  title?: string;
  author?: string;
  pictureId?: string;
  pictureUrl?: string;
  know?: number;
}

export class BookDto implements Book {
  constructor (
    public id?: number,
    public title?: string,
    public author?: string,
    public pictureId?: string,
    public pictureUrl?: string,
    public know?: number
  ) {
    this.pictureId = this.pictureId || ''
    this.pictureUrl = this.pictureUrl || ''
  }
}

File: src/main/java/ru/mrak/dto/BookDto.java
package ru.mrak.dto;

import lombok.Getter;
import lombok.Setter;
import lombok.ToString;
import ru.mrak.model.entity.Book;

import java.io.Serializable;
import java.util.Collection;

/**
 * A DTO for the {@link Book} entity.
 */
@Getter
@Setter
@ToString
public class BookDto implements Serializable {

    private Long id;
    private String title;
    private String author;
    private String pictureId;
    private Double know;
    private Boolean isReading;
    private Collection<TextTagDto> textTags;

}
